### PR TITLE
Enforce Ruff formatting where it was accidentally turned off

### PR DESCRIPTION
**Pull request by eerovaher**
### Description

[Ruff formatter is somewhat picky about the placement of format suppression instructions](https://docs.astral.sh/ruff/formatter/#format-suppression), which can cause the formatter to be accidentally turned off. The Ruff rule [RUF028 (invalid-formatter-suppression-comment)](https://docs.astral.sh/ruff/rules/invalid-formatter-suppression-comment/#invalid-formatter-suppression-comment-ruf028) guards against misplaced instructions, but it is not turned on in our configuration because it is still in [preview](https://docs.astral.sh/ruff/preview/). It is possible to look for misplaced instructions by running `ruff check --select RUF028 --preview` locally, which revealed problems in one file in `units`.

I also checked what our developer documentation had to say about suppressing the formatter and found that the description is factually wrong. Instead of updating the description it is better to link to Ruff documentation and let it provide the explanation.

- [x] By checking this box, the PR author has requested that maintainers do **NOT** use the "Squash and Merge" button. Maintainers should respect this when possible; however, the final decision is at the discretion of the maintainer that merges the PR.

### CODE REVIEW

File: astropy/units/quantity_helper/function_helpers.py [hunk 1]
```diff
@@ -469,7 +473,7 @@ def arange_impl(*args, start=None, stop=None, step=None, dtype=None, **kwargs):
     # bind positional arguments to their meaningful names
     # following the (complex) logic of np.arange
     match args:
-        case pos1,:
+        case (pos1,):
```
Comment (pllim) [inline astropy/units/quantity_helper/function_helpers.py:476]: Why this change?

**Review by neutrinoceros — APPROVED** [review]
Thanks a lot for digging into this ! I noticed that ruff didn't appear to do its job in the late stage of my recent patches, but didn't understand why and always had something more pressing on my plate.

**Review by pllim — APPROVED** [review]
Some of the rules still perplex me, but overall LGTM. I won't merge in case you need more input. Will let you decide when. Thanks!

File: astropy/units/quantity_helper/function_helpers.py (hunk 1 above)
Comment (eerovaher) [reply astropy/units/quantity_helper/function_helpers.py:476]: In this file I only replaced the two `fmt: off`/`fmt: on` pairs that RUF028 highlighted with `fmt: skip`. All the other changes were made automatically by Ruff formatter.

Comment (neutrinoceros) [reply astropy/units/quantity_helper/function_helpers.py:476]: I guess this is a ruff-format coming back to life.

For reference, the change is logically equivalent, but admittedly clearer: the comma by itself makes this expression a tuple, but most reader could think that it is a syntax error (and be confused that it runs), or  miss the comma entirely while reading and miss this branch for `case pos1` (which has the form of a catch-all `case`, which makes no sense as a first `case` since it would make all other branches unreachable).

Comment (pllim) [reply astropy/units/quantity_helper/function_helpers.py:476]: Last time I used such a thing, it was in C. So my brain cannot comprehend why it has to be a tuple with one element here. I find `case pos1` much easier to understand. 🤷‍♀️

Comment (neutrinoceros) [reply astropy/units/quantity_helper/function_helpers.py:476]: `case pos1:` means "match anything that can be binded to a variable named `pos1`. In other words, it matches anything.
`case (pos1,):` on the other hand roughly translates to "`if isinstance(x, Sequence) and len(x)==1`, then bind `x[0]` to `pos1`".
admittedly, Python 3.10's pattern matching takes some getting use to, and it's "brand new" for most of us who had to maintain compatibility with Python 3.9 until recently. I wouldn't have used it here if a classic `if/else` tree made the logic easier to parse.

**Review by nstarman — APPROVED** [review]
💯

File: astropy/units/quantity_helper/function_helpers.py (hunk 1 above)
Comment (pllim) [reply astropy/units/quantity_helper/function_helpers.py:476]: Oh... Indeed I did miss that trailing comma... 😆 😭 

Okay. The parenthesis is indeed better then. Thanks!

**Review by mhvk — APPROVED** [review]
Super! @neutrinoceros and I had indeed wondered why the pre-commit had not inserted a space after a `:` in a dict definition, but now it is clear why (and that we missed one too).

I also like just referring to the actual documentation about how to turn formatting off - it is still nice to avoid super long one-element-per-line sequences.
